R/DBConnection.R

```diff
@@ -106,6 +106,97 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' Sentinel value to indicate the early termination of
+#' \code{\link{dbGetChunkedQuery}}
+#' @export
+dbBreak <- structure(list(), class = "dbi_break_sentinel")
+
+dbFetchChunked <- function(rs, n, callback) {
+  while (TRUE) {
+    chunk <- dbFetch(rs, n = n)
+    if (nrow(chunk) == 0) {
+      return(chunk)
+    }
+    result <- callback(chunk)
```

Comment:
I wonder if it would be useful to include the cumulative number of rows in the callback? Or the number of times it's been called?

And I _think_, the call should be `callback(chunk = chunk)` to force people to use a consistently named argument. That  makes it easier to add new arguments in the future. But @jcheng5 will have better advice based on shiny experience.